### PR TITLE
Fix support for `enumType` on array fields

### LINKED ISSUE #478: The type integration does not properly support enumType on an array-based type
Currently, the resolution of doctrine types considers that any field that uses `enumType` will have a PHP type of `new ObjectType($enumType)` (potentially nullable depending on the nullability of the field).

This is a wrong assumption about the way this feature works in the Doctrine ORM. The `enumType` feature supports 2 cases:

- if the converted field value (based on the DBAL type of the field) is a scalar, it gets turned into an enum instance
- if the converted field value is an array, each value of the array gets turned into an enum instance

That second case is not supported by phpstan-doctrine, reporting the wrong type.
Both QueryResultTypeWalker and EntityColumnRule are impacted by that.

### PR DESCRIPTION
**Pull request by stof**
Closes https://github.com/phpstan/phpstan-doctrine/issues/478

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
Thank you!

**Comment by ondrejmirtes** [comment]
Could you also look into this? https://github.com/phpstan/phpstan-doctrine/issues/320

And also https://github.com/phpstan/phpstan-doctrine/issues/509 + https://github.com/phpstan/phpstan-doctrine/issues/353

Thanks.

File: src/Type/Doctrine/Query/QueryResultTypeWalker.php [hunk 1]
```diff
@@ -2028,7 +2041,7 @@ private function resolveDoctrineType(string $typeName, ?string $enumType = null,
 			$type = TypeCombinator::addNull($type);
 		}
 
-		return $type;
+			return $type;
```
Comment (janedbal) [inline src/Type/Doctrine/Query/QueryResultTypeWalker.php:2044]: Looks like some cs check is missing :)

Comment (stof) [reply src/Type/Doctrine/Query/QueryResultTypeWalker.php:2044]: this was actually *introduced* by phpcbf

**Comment by stof** [comment]
@ondrejmirtes I commented on those 3 issues (and 2 of them are not actually related to the support of enums)

**Pull request merged**